app/models/zone.rb

```diff
@@ -201,7 +201,7 @@ def ems_clouds
 
   # @return [Array<ExtManagementSystem>] All emses that can collect Capacity and Utilization metrics
   def ems_metrics_collectable
-    ext_management_systems.select { |e| e.kind_of?(EmsCloud) || e.kind_of?(EmsInfra) || e.kind_of?(ManageIQ::Providers::ContainerManager) }
+    ext_management_systems.where(:type => ExtManagementSystem.subclasses_support?(:metrics).map(&:name))
```

Comment:
NOTE: this only checks that the class supports metrics, it is possible in the future that providers could implement a `supports?(:metrics)` instance block (e.g. checking if metrics credentials are present/valid).

Another approach we could take would be `ext_management_systems.select { |e| e.supports?(:metrics) }` would support that but with the performance penalty of instantiating every instance to check (guess we were doing that already though?)

Reply:
We could combine the class and instance checks, do a `.where(:type)` then `.select { |e| e.supports?(:metrics) }` so we only instantiate instances which could possibly support metrics

Reply:
I like the combination check.  It doesn't return a scope, unfortunately, but currently all of the callers immediately call .each anyway, so it should make a difference.